solarforecastarbiter/reference_forecasts/models.py

```diff
@@ -0,0 +1,165 @@
+def hrrr_subhourly_to_hourly_mean(latitude, longitude, elevation):
+    """
+    Hourly mean HRRR forecast.
+    """
+    ghi, dni, dhi, temp_air, wind_speed = load_forecast(
+        latitude, longitude, 'hrrr_subhourly')
+    resampler = partial(forecast.resample, freq='1h')
+    return None, None, ghi, dni, dhi, temp_air, wind_speed, resampler
+
+
+def rap_to_instantaneous(latitude, longitude, elevation):
+    """
+    Hourly instantantaneous RAP forecast.
+    """
+    ghi, dni, dhi, temp_air, wind_speed = load_forecast(latitude, longitude,
+                                                        'rap')
+    resampler = partial(forecast.resample, freq='1h')
+    return None, None, ghi, dni, dhi, temp_air, wind_speed, resampler
+
+
+def rap_irrad_to_hourly_mean(latitude, longitude, elevation):
+    """
+    Take hourly RAP instantantaneous irradiance and convert it to hourly
+    average data.
+    """
+    ghi, dni, dhi, temp_air, wind_speed = load_forecast(
+        latitude, longitude, 'rap')
+    resample = partial(forecast.resample, freq='1h')
+    return None, None, ghi, dni, dhi, temp_air, wind_speed, resample
+
+
+def rap_cloud_cover_to_hourly_mean(latitude, longitude, elevation):
+    """
+    Take hourly RAP instantantaneous cloud cover and convert it to
+    hourly average data.
+    """
+    variables = load_forecast(
+        latitude, longitude, 'rap',
+        variables=('cloud_cover', 'temp_air', 'wind_speed'))
+    interpolator = partial(forecast.interpolate, freq='15min')
+    cloud_cover, temp_air, wind_speed = list(map(interpolator, variables))
+    solpos_irrad = forecast.cloud_cover_to_irradiance_clearsky_scaling_solpos(
+            latitude, longitude, elevation, cloud_cover)
+    resampler = partial(forecast.resample, freq='1h')
+    return solpos_irrad + (temp_air, wind_speed, resampler)
+
+
+def rap_cloud_cover_to_hourly_mean_alternative(latitude, longitude, elevation,
+                                               solar_position_func):
+    """
+    Take hourly RAP instantantaneous cloud cover and convert it to
+    hourly average data.
+    """
+    variables = load_forecast(
+        latitude, longitude, 'rap',
+        variables=('cloud_cover', 'temp_air', 'wind_speed'))
+    resampler = partial(forecast.resample, freq='15')
+    cloud_cover, temp_air, wind_speed = list(map(resampler, variables))
+    solar_position = solar_position_func(cloud_cover.index)
+    cs = pvmodel.calculate_clearsky(latitude, longitude, elevation,
+                                    solar_position['apparent_zenith'])
+    ghi, dni, dhi = forecast.cloud_cover_to_irradiance_clearsky_scaling(
+        cloud_cover, cs['ghi'], solar_position['zenith']
+    )
+    resampler = partial(forecast.resample, freq='1h')
+    solpos = solar_position['apparent_zenith'], solar_position['azimuth']
+    return solpos + (ghi, dni, dhi, temp_air, wind_speed, resampler)
+
+
+def rap_cloud_cover_to_hourly_mean_alternative2(latitude, longitude, elevation,
```

Comment:
my personal choice. solar_position_func need not include latitude, longitude, elevation, and maybe shouldn't for clarity

Reply:
then it's just ``model(latitude, longitude, elevation, calculate_solar_position)``

Reply:
If I understand correctly, the only benefit to that pattern is the possibility that `calculate_solar_position` would be decorated with a caching function at a higher level. Is that right?

Reply:
I should say "could be" since it's TBD if it's feasible.

Reply:
and the possible benefit that we think twice before throwing it around in lower code all willy  nilly

Reply:
we can think twice about it without making the code unnecessarily complicated.

here's another idea:

```python
def myfxmodel_that_doesnt_need_solpos(latitude, longitude, elevation):
    load_forecast, etc.
    solar_position_calculator = partial(pvmodel.calculate_solar_position, latitude, longitude, elevation)
    return ghi, dni, dhi, temp_air, wind_speed, resampler, solar_position_calculator


def myfxmodel_that_needs_solpos(latitude, longitude, elevation):
    load_forecast, etc.
    solar_position = pvmodel.calculate_solar_position(latitude, longitude, elevation)
    def solar_position_calculator: return solar_position
    return ghi, dni, dhi, temp_air, wind_speed, resampler, solar_position_calculator